File: PresentationGenerator.m
curdir = 'C:\Users\mgorbun\Documents\Presentations\';
figDir = 'fig';
outFile = fullfile(curdir, 'slides1.md');

ext = {'*.png','*.jpg','*.jpeg','*.tif','*.tiff','*.bmp'};

files = [];
for k = 1:numel(ext)
    files = [files; dir(fullfile(curdir, figDir, ext{k}))];
end

fid = fopen(outFile,'w');

for k = 1:numel(files)
    name = files(k).name;
    % fprintf(fid,'#\n');
    fprintf(fid,'# %s\n\n', name(1:3));
    % fprintf(fid,'---\n');    
    % fprintf(fid,'layout: blank \n');
    % fprintf(fid,'---\n');
    fprintf(fid,'![](%s/%s){width=6cm}\n\n', figDir, name);

    if k < numel(files)
        fprintf(fid,'---\n\n');
    end
end

fclose(fid);


%%

curdir = 'C:\Users\mgorbun\Documents\Presentations\';
figDir = 'fig';
outFile = fullfile(curdir, 'slides1.org');

ext = {'*.png','*.jpg','*.jpeg'}%,'*.tif','*.tiff','*.bmp'};

files = [];
for k = 1:numel(ext)
    files = [files; dir(fullfile(curdir, figDir, ext{k}))];
end

fid = fopen(outFile,'w');

fprintf(fid,'#+TITLE: Slides\n\n');

for k = 1:numel(files)
    name = files(k).name;

    title = name(1:min(3,numel(name)));

    % fprintf(fid,'* %s\n\n', title);
    fprintf(fid,'#+ATTR_PANDOC: :width 16cm\n');
    fprintf(fid,'[[file:%s/%s]]\n\n', figDir, name);
end

fclose(fid);
disp('That''s it')

%%

curdir = 'C:\Users\mgorbun\Documents\Presentations\fig';

files = dir(fullfile(curdir,'*.tif'));

for k = 1:numel(files)
    name = files(k).name;

    I = imread(fullfile(curdir,name));
    imwrite(I, fullfile(curdir, [name(1:end-4) '.png']));
end

disp('1')

File: PresentationGenerator.py
from pptx import Presentation
from pptx.enum.text import MSO_AUTO_SIZE
from pptx.util import Cm
from pptx.util import Pt
from PIL import Image
import os

fig_dir = r"C:\Users\mgorbun\Documents\Presentations\fig"
out_file = r"C:\Users\mgorbun\Documents\Presentations\auto.pptx"

prs = Presentation()
blank = prs.slide_layouts[6]

for fname in sorted(os.listdir(fig_dir)):
    if fname.lower().endswith((".png",".jpg",".jpeg",".tif",".tiff",".bmp")):

        slide = prs.slides.add_slide(blank)
        path = os.path.join(fig_dir, fname)

        # желаемая ширина
        width = Cm(23)

        # вычисляем пропорциональную высоту
        img = Image.open(path)
        w_px, h_px = img.size
        height = width * h_px / w_px

        # центрирование
        left = (prs.slide_width - width) / 1.1
        top  = (prs.slide_height - height) / 1.1

        slide.shapes.add_picture(path, left, top, width=width)
        
        txBox = slide.shapes.add_textbox(Cm(0), Cm(0), Cm(1), Cm(1))  # размер пока не важен

        tf = txBox.text_frame
        tf.text = fname[:3]
        
        p = tf.paragraphs[0]
        p.font.size = Pt(40)
        
        tf.word_wrap = False
        tf.auto_size = MSO_AUTO_SIZE.SHAPE_TO_FIT_TEXT

prs.save(out_file)

print('That\'s it')